src/vdom/component.js

```diff
@@ -105,40 +118,49 @@ export function renderComponent(component, opts, mountAll, isChild) {
 		let childComponent = rendered && rendered.nodeName,
 			toUnmount, base;
 
-		if (typeof childComponent==='function') {
-			// set up high order component link
+		try {
```

Comment:
This is the try/catch I was referring to in my other review comment.

We can/should call `catchErrorInComponent` from within here directly to avoid the extra layer(s) of try/catch blocks.